store/statistics.js

```diff
@@ -0,0 +1,43 @@
+import { action, extendObservable } from 'mobx'
+
+import apiRequest from 'api'
+
+const initialStore = {
+  metadataCount: 0,
+  fullTextCount: 0,
+  doiCount: 0,
+}
+
+class Statistics {
+  constructor(rootStore) {
+    this.rootStore = rootStore
+    extendObservable(this, initialStore)
+  }
+
+  retrieveStatistics = async (dataProviderId = this.rootStore.dataProvider) => {
+    try {
+      const { data } = await apiRequest(
+        `/data-providers/${dataProviderId}/statistics`,
+        'GET',
+        {},
+        null,
+        true
+      )
+
+      return data || {}
+    } catch (e) {
+      if (e.statusCode !== 404) throw e
+      return {}
+    }
+  }
+
+  @action
+  onDataProviderChange = async (
+    dataProviderId = this.rootStore.dataProvider
+  ) => {
+    const statistics = await this.retrieveStatistics(dataProviderId)
+    Object.assign(this, statistics)
```

Comment:
@Joozty I saw you used similar approach. Please, check in line 14 how I solved the issue with this. Line from the MobX's documentation:
> Only plain objects will be made observable. For non-plain objects it is considered the responsibility of the constructor to initialize the observable properties. Either use the `@observable` annotation or the `extendObservable` function.